From 872398dbd3257f44a1da4bc894130f01dd4ebeb6 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 30 Oct 2017 15:25:11 +0800
Subject: [PATCH 2723/5242] MLK-16686-2 ARM64: dts: imx8mq: enable rpmsg on
 imx8mq

commit  6fbb121d0b3816049802f73ace8cea00dbb42158 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable rpmsg support on imx8mq platforms.
- To avoid the potenial confliction, reduce the allocation
scope of CMA.
- Reserved the 0xb800_0000 1Mbyte for RPMSG

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |   14 ++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi    |   20 ++++++++++++++++++--
 2 files changed, 32 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 7116502..1675216 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -644,3 +644,17 @@
 	fsl,ext-reset-output;
 	status = "okay";
 };
+
+&mu {
+	status = "okay";
+};
+
+&rpmsg{
+	/*
+	 * 64K for one rpmsg instance:
+	 * --0xb8000000~0xb800ffff: pingpong
+	 */
+	vdev-nums = <1>;
+	reg = <0x0 0xb8000000 0x0 0x10000>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 653ed59..5613f8e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -57,9 +57,14 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			size = <0 0x28000000>;
-			alloc-ranges = <0 0x40000000 0 0x80000000>;
+			alloc-ranges = <0 0x40000000 0 0x70000000>;
 			linux,cma-default;
 		};
+
+		rpmsg_reserved: rpmsg@0xb8000000 {
+			no-map;
+			reg = <0 0xb8000000 0 0x100000>;
+		};
 	};
 
 	gic: interrupt-controller@38800000 {
@@ -710,7 +715,7 @@
 		compatible = "fsl,imx8mq-mu", "fsl,imx6sx-mu";
 		reg = <0x0 0x30aa0000 0x0 0x10000>;
 		interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&clk IMX8MQ_CLK_DUMMY>;
+		clocks = <&clk IMX8MQ_CLK_MU_ROOT>;
 		clock-names = "mu";
 		status = "disabled";
 	};
@@ -1151,6 +1156,17 @@
 		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
 	};
 
+	imx_rpmsg: imx_rpmsg {
+		compatible = "fsl,rpmsg-bus", "simple-bus";
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		rpmsg: rpmsg{
+			compatible = "fsl,imx8qm-rpmsg";
+			status = "disabled";
+		};
+	};
 };
 
 &A53_0 {
-- 
1.7.9.5

